src/com/linkedin/parseq/BaseTask.java

```diff
@@ -122,7 +125,13 @@ public BaseTask(final String name, final String taskType) {
     if (taskType != null) {
       _shallowTraceBuilder.setTaskType(taskType);
     }
-    _stateRef = new AtomicReference<State>(state);
+    _stateRef = new AtomicReference<>(state);
+
+    if (ParSeqGlobalConfiguration.getInstance().isCrossThreadStackTracesEnabled()) {
```

Comment:
ParSeqGlobalConfiguration.getInstance().isCrossThreadStackTracesEnabled() looks a bit verbose. I think that maybe having static methods in ParSeqGlobalConfiguration directly is a better idea after all.